src/v/security/audit/audit_log_manager.h

```diff
@@ -232,10 +226,50 @@ class audit_log_manager
     ss::future<> resume();
 
     bool is_audit_event_enabled(event_type) const;
-    bool do_enqueue_audit_event(
-      std::unique_ptr<security::audit::ocsf_base_impl> msg);
     void set_enabled_events();
 
+    template<typename T, typename... Args>
+    bool do_enqueue_audit_event(Args&&... args) {
+        auto& map = _queue.get<underlying_unordered_map>();
+        const auto hash_key = T::hash(args...);
```

Comment:
Would it not make sense to reuse an existing hashing extension point such as `std::hash`, `boost::hash`, or `absl::Hash`, rather than inventing another one?

Reply:
The difference here is that hash is a static method, we are calculating the hash at a point where the object doesn't yet exist.